extensions/objects/templates/ratio-expression-editor.component.spec.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the ratio expression component.
+ */
+
+import { RatioObjectFactory } from 'domain/objects/RatioObjectFactory';
+
+
+describe('RatioExpression', function() {
+  var RationExpressionCtrl = null;
+  var ratioObjectFactory = null;
+  beforeEach(angular.mock.module('oppia'));
+  beforeEach(angular.mock.module('oppia', function($provide) {
+    ratioObjectFactory = new RatioObjectFactory();
+    $provide.value('RatioObjectFactory', ratioObjectFactory);
+  }));
+  beforeEach(angular.mock.inject(function($componentController) {
+    RationExpressionCtrl = $componentController('ratioExpressionEditor');
+  }));
+
+  it('should initialize ctrl.value with an default value', function() {
+    RationExpressionCtrl.value = null;
+    RationExpressionCtrl.$onInit();
+    expect(RationExpressionCtrl.value).not.toBeNull();
+  });
+
+  it('should initialize ctrl.warningText with non integer ratio', function() {
```

Comment:
```suggestion
  it('should initialize ctrl.warningText with a non-integer ratio', function() {
```

Ditto elsewhere for the test wording.

Reply:
done